``` r
library(ShortRead)
```

    ## Loading required package: BiocGenerics

    ## Loading required package: parallel

    ## 
    ## Attaching package: 'BiocGenerics'

    ## The following objects are masked from 'package:parallel':
    ## 
    ##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    ##     clusterExport, clusterMap, parApply, parCapply, parLapply,
    ##     parLapplyLB, parRapply, parSapply, parSapplyLB

    ## The following objects are masked from 'package:stats':
    ## 
    ##     IQR, mad, sd, var, xtabs

    ## The following objects are masked from 'package:base':
    ## 
    ##     anyDuplicated, append, as.data.frame, basename, cbind,
    ##     colMeans, colnames, colSums, dirname, do.call, duplicated,
    ##     eval, evalq, Filter, Find, get, grep, grepl, intersect,
    ##     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
    ##     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
    ##     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
    ##     table, tapply, union, unique, unsplit, which, which.max,
    ##     which.min

    ## Loading required package: BiocParallel

    ## Loading required package: Biostrings

    ## Loading required package: S4Vectors

    ## Loading required package: stats4

    ## 
    ## Attaching package: 'S4Vectors'

    ## The following object is masked from 'package:base':
    ## 
    ##     expand.grid

    ## Loading required package: IRanges

    ## Loading required package: XVector

    ## 
    ## Attaching package: 'Biostrings'

    ## The following object is masked from 'package:base':
    ## 
    ##     strsplit

    ## Loading required package: Rsamtools

    ## Loading required package: GenomeInfoDb

    ## Loading required package: GenomicRanges

    ## Loading required package: GenomicAlignments

    ## Loading required package: SummarizedExperiment

    ## Loading required package: Biobase

    ## Welcome to Bioconductor
    ## 
    ##     Vignettes contain introductory material; view with
    ##     'browseVignettes()'. To cite Bioconductor, see
    ##     'citation("Biobase")', and for packages 'citation("pkgname")'.

    ## Loading required package: DelayedArray

    ## Loading required package: matrixStats

    ## 
    ## Attaching package: 'matrixStats'

    ## The following objects are masked from 'package:Biobase':
    ## 
    ##     anyMissing, rowMedians

    ## 
    ## Attaching package: 'DelayedArray'

    ## The following objects are masked from 'package:matrixStats':
    ## 
    ##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges

    ## The following object is masked from 'package:Biostrings':
    ## 
    ##     type

    ## The following objects are masked from 'package:base':
    ## 
    ##     aperm, apply

``` r
library(dada2)
```

    ## Loading required package: Rcpp

``` r
library(phyloseq)
```

    ## 
    ## Attaching package: 'phyloseq'

    ## The following object is masked from 'package:SummarizedExperiment':
    ## 
    ##     distance

    ## The following object is masked from 'package:Biobase':
    ## 
    ##     sampleNames

    ## The following object is masked from 'package:GenomicRanges':
    ## 
    ##     distance

    ## The following object is masked from 'package:IRanges':
    ## 
    ##     distance

Set to directory with raw reads

``` r
setwd("~/Desktop/Ocean-Crust-Synthesis/SRP063681//")
```

``` r
path <- "~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs/"
list.files(path)
```

    ##  [1] "Blank-1_R1.fq" "Blank-1_R2.fq" "Blank-2_R1.fq" "Blank-2_R2.fq"
    ##  [5] "Blank-3_R1.fq" "Blank-3_R2.fq" "Blank-4_R1.fq" "Blank-4_R2.fq"
    ##  [9] "BW-1_R1.fq"    "BW-1_R2.fq"    "BW-2_R1.fq"    "BW-2_R2.fq"   
    ## [13] "filtered"      "R1_R1.fq"      "R1_R2.fq"      "R10_R1.fq"    
    ## [17] "R10_R2.fq"     "R11_R1.fq"     "R11_R2.fq"     "R11-BF_R1.fq" 
    ## [21] "R11-BF_R2.fq"  "R12_R1.fq"     "R12_R2.fq"     "R1A_R1.fq"    
    ## [25] "R1A_R2.fq"     "R2_R1.fq"      "R2_R2.fq"      "R3_R1.fq"     
    ## [29] "R3_R2.fq"      "R4_R1.fq"      "R4_R2.fq"      "R5_R1.fq"     
    ## [33] "R5_R2.fq"      "R6_R1.fq"      "R6_R2.fq"      "R7_R1.fq"     
    ## [37] "R7_R2.fq"      "R8_R1.fq"      "R8_R2.fq"      "R9_R1.fq"     
    ## [41] "R9_R2.fq"

``` r
# Sort ensures forward/reverse reads are in same order
fnFs <- sort(list.files(path, pattern="_R1.fq"))
fnRs <- sort(list.files(path, pattern="_R2.fq"))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(fnFs, "_"), `[`, 1)
# Specify the full path to the fnFs and fnRs
fnFs <- file.path(path, fnFs)
fnRs <- file.path(path, fnRs)
```

``` r
sample.names
```

    ##  [1] "Blank-1" "Blank-2" "Blank-3" "Blank-4" "BW-1"    "BW-2"    "R1"     
    ##  [8] "R10"     "R11"     "R11-BF"  "R12"     "R1A"     "R2"      "R3"     
    ## [15] "R4"      "R5"      "R6"      "R7"      "R8"      "R9"

``` r
filt_path <- file.path(path, "filtered") # Place filtered files in filtered/ subdirectory
filtFs <- file.path(filt_path, paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sample.names, "_R_filt.fastq.gz"))
```

``` r
plotQualityProfile(fnFs[1:2])
```

    ## Scale for 'y' is already present. Adding another scale for 'y', which
    ## will replace the existing scale.

![](SRP06381-Lee18_files/figure-markdown_github/unnamed-chunk-7-1.png)

``` r
plotQualityProfile(fnRs[1:2])
```

    ## Scale for 'y' is already present. Adding another scale for 'y', which
    ## will replace the existing scale.

![](SRP06381-Lee18_files/figure-markdown_github/unnamed-chunk-8-1.png)

Trim off primed region, trimleft=19.

``` r
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(250,200), trimLeft = 19,
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
head(out)
```

    ##               reads.in reads.out
    ## Blank-1_R1.fq    16718     15647
    ## Blank-2_R1.fq     6129      5623
    ## Blank-3_R1.fq     5273      4856
    ## Blank-4_R1.fq     5244      4919
    ## BW-1_R1.fq       23803     22062
    ## BW-2_R1.fq       63606     58982

``` r
errF1 <- learnErrors(filtFs, multithread=TRUE)
```

    ## 120044001 total bases in 519671 reads from 10 samples will be used for learning the error rates.
    ## Initializing error rates to maximum possible estimate.
    ## selfConsist step 1 ..........
    ##    selfConsist step 2
    ##    selfConsist step 3
    ##    selfConsist step 4
    ##    selfConsist step 5
    ##    selfConsist step 6
    ## Convergence after  6  rounds.

``` r
errR1 <- learnErrors(filtRs, multithread=TRUE)
```

    ## 121847933 total bases in 673193 reads from 11 samples will be used for learning the error rates.
    ## Initializing error rates to maximum possible estimate.
    ## selfConsist step 1 ...........
    ##    selfConsist step 2
    ##    selfConsist step 3
    ##    selfConsist step 4
    ##    selfConsist step 5
    ##    selfConsist step 6
    ## Convergence after  6  rounds.

``` r
plotErrors(errF1, nominalQ=TRUE)
```

    ## Warning: Transformation introduced infinite values in continuous y-axis

    ## Warning: Transformation introduced infinite values in continuous y-axis

![](SRP06381-Lee18_files/figure-markdown_github/unnamed-chunk-11-1.png)

``` r
plotErrors(errR1, nominalQ=TRUE)
```

    ## Warning: Transformation introduced infinite values in continuous y-axis

    ## Warning: Transformation introduced infinite values in continuous y-axis

![](SRP06381-Lee18_files/figure-markdown_github/unnamed-chunk-12-1.png)

``` r
derepFs <- derepFastq(filtFs, verbose=TRUE)
```

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-1_F_filt.fastq.gz

    ## Encountered 3864 unique sequences from 15647 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-2_F_filt.fastq.gz

    ## Encountered 1626 unique sequences from 5623 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-3_F_filt.fastq.gz

    ## Encountered 1505 unique sequences from 4856 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-4_F_filt.fastq.gz

    ## Encountered 1536 unique sequences from 4919 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/BW-1_F_filt.fastq.gz

    ## Encountered 5513 unique sequences from 22062 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/BW-2_F_filt.fastq.gz

    ## Encountered 20470 unique sequences from 58982 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R1_F_filt.fastq.gz

    ## Encountered 56204 unique sequences from 118328 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R10_F_filt.fastq.gz

    ## Encountered 41946 unique sequences from 112894 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R11_F_filt.fastq.gz

    ## Encountered 37914 unique sequences from 87605 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R11-BF_F_filt.fastq.gz

    ## Encountered 25587 unique sequences from 88755 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R12_F_filt.fastq.gz

    ## Encountered 77847 unique sequences from 153522 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R1A_F_filt.fastq.gz

    ## Encountered 73539 unique sequences from 157599 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R2_F_filt.fastq.gz

    ## Encountered 74380 unique sequences from 168384 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R3_F_filt.fastq.gz

    ## Encountered 80900 unique sequences from 170908 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R4_F_filt.fastq.gz

    ## Encountered 79258 unique sequences from 183058 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R5_F_filt.fastq.gz

    ## Encountered 87848 unique sequences from 180251 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R6_F_filt.fastq.gz

    ## Encountered 68064 unique sequences from 144801 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R7_F_filt.fastq.gz

    ## Encountered 32419 unique sequences from 77912 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R8_F_filt.fastq.gz

    ## Encountered 48076 unique sequences from 121586 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R9_F_filt.fastq.gz

    ## Encountered 31716 unique sequences from 85800 total sequences read.

``` r
derepRs <- derepFastq(filtRs, verbose=TRUE)
```

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-1_R_filt.fastq.gz

    ## Encountered 3709 unique sequences from 15647 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-2_R_filt.fastq.gz

    ## Encountered 1532 unique sequences from 5623 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-3_R_filt.fastq.gz

    ## Encountered 1398 unique sequences from 4856 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/Blank-4_R_filt.fastq.gz

    ## Encountered 1421 unique sequences from 4919 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/BW-1_R_filt.fastq.gz

    ## Encountered 5224 unique sequences from 22062 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/BW-2_R_filt.fastq.gz

    ## Encountered 18692 unique sequences from 58982 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R1_R_filt.fastq.gz

    ## Encountered 52799 unique sequences from 118328 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R10_R_filt.fastq.gz

    ## Encountered 39355 unique sequences from 112894 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R11_R_filt.fastq.gz

    ## Encountered 36172 unique sequences from 87605 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R11-BF_R_filt.fastq.gz

    ## Encountered 23827 unique sequences from 88755 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R12_R_filt.fastq.gz

    ## Encountered 73337 unique sequences from 153522 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R1A_R_filt.fastq.gz

    ## Encountered 68860 unique sequences from 157599 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R2_R_filt.fastq.gz

    ## Encountered 69041 unique sequences from 168384 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R3_R_filt.fastq.gz

    ## Encountered 75685 unique sequences from 170908 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R4_R_filt.fastq.gz

    ## Encountered 75372 unique sequences from 183058 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R5_R_filt.fastq.gz

    ## Encountered 83646 unique sequences from 180251 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R6_R_filt.fastq.gz

    ## Encountered 65030 unique sequences from 144801 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R7_R_filt.fastq.gz

    ## Encountered 30794 unique sequences from 77912 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R8_R_filt.fastq.gz

    ## Encountered 45607 unique sequences from 121586 total sequences read.

    ## Dereplicating sequence entries in Fastq file: ~/Desktop/Ocean-Crust-Synthesis/SRP063681/dorado_split_fqs//filtered/R9_R_filt.fastq.gz

    ## Encountered 30059 unique sequences from 85800 total sequences read.

``` r
# Name the derep-class objects by the sample names
names(derepFs) <- sample.names
names(derepRs) <- sample.names
```

``` r
dadaFs <- dada(derepFs, err=errF1, multithread=TRUE)
```

    ## Sample 1 - 15647 reads in 3864 unique sequences.
    ## Sample 2 - 5623 reads in 1626 unique sequences.
    ## Sample 3 - 4856 reads in 1505 unique sequences.
    ## Sample 4 - 4919 reads in 1536 unique sequences.
    ## Sample 5 - 22062 reads in 5513 unique sequences.
    ## Sample 6 - 58982 reads in 20470 unique sequences.
    ## Sample 7 - 118328 reads in 56204 unique sequences.
    ## Sample 8 - 112894 reads in 41946 unique sequences.
    ## Sample 9 - 87605 reads in 37914 unique sequences.
    ## Sample 10 - 88755 reads in 25587 unique sequences.
    ## Sample 11 - 153522 reads in 77847 unique sequences.
    ## Sample 12 - 157599 reads in 73539 unique sequences.
    ## Sample 13 - 168384 reads in 74380 unique sequences.
    ## Sample 14 - 170908 reads in 80900 unique sequences.
    ## Sample 15 - 183058 reads in 79258 unique sequences.
    ## Sample 16 - 180251 reads in 87848 unique sequences.
    ## Sample 17 - 144801 reads in 68064 unique sequences.
    ## Sample 18 - 77912 reads in 32419 unique sequences.
    ## Sample 19 - 121586 reads in 48076 unique sequences.
    ## Sample 20 - 85800 reads in 31716 unique sequences.

``` r
dadaRs <- dada(derepRs, err=errR1, multithread=TRUE)
```

    ## Sample 1 - 15647 reads in 3709 unique sequences.
    ## Sample 2 - 5623 reads in 1532 unique sequences.
    ## Sample 3 - 4856 reads in 1398 unique sequences.
    ## Sample 4 - 4919 reads in 1421 unique sequences.
    ## Sample 5 - 22062 reads in 5224 unique sequences.
    ## Sample 6 - 58982 reads in 18692 unique sequences.
    ## Sample 7 - 118328 reads in 52799 unique sequences.
    ## Sample 8 - 112894 reads in 39355 unique sequences.
    ## Sample 9 - 87605 reads in 36172 unique sequences.
    ## Sample 10 - 88755 reads in 23827 unique sequences.
    ## Sample 11 - 153522 reads in 73337 unique sequences.
    ## Sample 12 - 157599 reads in 68860 unique sequences.
    ## Sample 13 - 168384 reads in 69041 unique sequences.
    ## Sample 14 - 170908 reads in 75685 unique sequences.
    ## Sample 15 - 183058 reads in 75372 unique sequences.
    ## Sample 16 - 180251 reads in 83646 unique sequences.
    ## Sample 17 - 144801 reads in 65030 unique sequences.
    ## Sample 18 - 77912 reads in 30794 unique sequences.
    ## Sample 19 - 121586 reads in 45607 unique sequences.
    ## Sample 20 - 85800 reads in 30059 unique sequences.

``` r
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
```

    ## 15347 paired-reads (in 54 unique pairings) successfully merged out of 15412 (in 75 pairings) input.

    ## 5536 paired-reads (in 22 unique pairings) successfully merged out of 5539 (in 24 pairings) input.

    ## 4707 paired-reads (in 23 unique pairings) successfully merged out of 4710 (in 26 pairings) input.

    ## 4695 paired-reads (in 36 unique pairings) successfully merged out of 4742 (in 45 pairings) input.

    ## 21309 paired-reads (in 90 unique pairings) successfully merged out of 21495 (in 173 pairings) input.

    ## 54382 paired-reads (in 564 unique pairings) successfully merged out of 56674 (in 1256 pairings) input.

    ## 97527 paired-reads (in 1494 unique pairings) successfully merged out of 109176 (in 5283 pairings) input.

    ## 101943 paired-reads (in 1190 unique pairings) successfully merged out of 108962 (in 3282 pairings) input.

    ## 74175 paired-reads (in 976 unique pairings) successfully merged out of 82138 (in 3509 pairings) input.

    ## 80986 paired-reads (in 650 unique pairings) successfully merged out of 85092 (in 2082 pairings) input.

    ## 119321 paired-reads (in 1772 unique pairings) successfully merged out of 139393 (in 8084 pairings) input.

    ## 129480 paired-reads (in 1921 unique pairings) successfully merged out of 145714 (in 7149 pairings) input.

    ## 143410 paired-reads (in 1984 unique pairings) successfully merged out of 157456 (in 6840 pairings) input.

    ## 141699 paired-reads (in 1908 unique pairings) successfully merged out of 159139 (in 7564 pairings) input.

    ## 163007 paired-reads (in 1760 unique pairings) successfully merged out of 177011 (in 6092 pairings) input.

    ## 145999 paired-reads (in 2054 unique pairings) successfully merged out of 166598 (in 8172 pairings) input.

    ## 121556 paired-reads (in 1829 unique pairings) successfully merged out of 135290 (in 6149 pairings) input.

    ## 70277 paired-reads (in 1423 unique pairings) successfully merged out of 74552 (in 2884 pairings) input.

    ## 109271 paired-reads (in 1304 unique pairings) successfully merged out of 116296 (in 3745 pairings) input.

    ## 74674 paired-reads (in 879 unique pairings) successfully merged out of 80812 (in 2889 pairings) input.

``` r
# Inspect the merger data.frame from the first sample
head(mergers[[1]])
```

    ##                                                                                                                                                                                                                                                         sequence
    ## 1  TACGTAGGTCCCGAGCGTTGTCCGGATTTATTGGGCGTAAAGCGAGCGCAGGTGGTTTATTAAGTCTGGTGTAAAAGGCAGTGGCTCAACCATTGTATGCATTGGAAACTGGTAGACTTGAGTGCAGGAGAGGAGAGTGGAATTCCATGTGTAGCGGTGAAATGCGTAGATATATGGAGGAACACCGGTGGCGAAAGCGGCTCTCTGGCCTGTAACTGACACTGAGGCTCGAAAGCGTGGGGAGCAAACAGGA
    ## 2 TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTTTGTAAGTCTGACGTGAAATCCCCGGGCTCAACCTGGGAATTGCGTTGGAGACTGCAAGGCTAGAATCTGGCAGAGGGGGGTAGAATTCCACGTGTAGCAGTGAAATGCGTAGAGATGTGGAGGAACACCGATGGCGAAGGCAGCCCCCTGGGTCAAGATTGACGCTCATGCACGAAAGCGTGGGGAGCAAACAGGA
    ## 3 TACGGAGGGTGCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTCCGTAGGCGGACTTATAAGTCAGTGGTGAAAGCCTGTCGCTTAACGATAGAACTGCCATTGATACTGTAAGTCTTGAGTATATTTGAGGTAGCTGGAATAAGTAGTGTAGCGGTGAAATGCATAGATATTACTTAGAACACCAATTGCGAAGGCAGGTTACCAAGATATAACTGACGCTGAGGGACGAAAGCGTGGGGAGCGAACAGGA
    ## 4 TACGGAGGGTGCAAGCGTTATCCGGATTCATTGGGTTTAAAGGGTCCGCAGGCGGGCTTGTAAGTCAGGGGTGAAAGCCGGCAGCTTAACTGTCGAACTGCCTTTGATACTGCAAGTCTTGAGTGAGGTTGAAGTAGCTGGAATGTGTAGTGTAGCGGTGAAATGCTTAGATATTACACAGAACACCGATAGCGAAGGCAGGTTACTAAGTCTTAACTGACGCTGAGGGACGAAAGCGTGGGGAGCGAACAGGA
    ## 5 TACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTCTTGTAAGACAGAGGTGAAATCCCTGGGCTCAACCTAGGAATGGCCTTTGTGACTGCAAGGCTGGAGTGCGGCAGAGGGGGATGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAGGAACACCGATGGCGAAGGCAGTCCCCTGGGCCTGCACTGACGCTCATGCACGAAAGCGTGGGGAGCAAACAGGA
    ## 6 TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTCTGTCAAGTCGGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTCGAAACTGGCAGGCTAGAGTCTTGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGA
    ##   abundance forward reverse nmatch nmismatch nindel prefer accept
    ## 1      2182       1       1    159         0      0      2   TRUE
    ## 2      1640       3       2    158         0      0      2   TRUE
    ## 3      1626       2       3    158         0      0      2   TRUE
    ## 4      1382       4       4    158         0      0      2   TRUE
    ## 5      1280       5       5    158         0      0      2   TRUE
    ## 6       889       7       7    158         0      0      2   TRUE

``` r
seqtab <- makeSequenceTable(mergers)
```

    ## The sequences being tabled vary in length.

``` r
dim(seqtab)
```

    ## [1]    20 11321

``` r
table(nchar(getSequences(seqtab)))
```

    ## 
    ##   231   233   234   235   236   237   244   249   250   252   253   254 
    ##    18     2   298    29     1    24     1     1     1     4    63 10269 
    ##   255   256   257   258   259   263   270   278   279   283   284   306 
    ##   562     9     4     1     1     3     1     1     1     1     1     1 
    ##   347   348   349   350   351   364   365   366   367   368   373   381 
    ##     1     1     2     2     1     1     1     5     4     1     2     1 
    ##   396   399 
    ##     1     1

``` r
seqtab2 <- seqtab[,nchar(colnames(seqtab)) %in% seq(250,255)]
dim(seqtab2)
```

    ## [1]    20 10899

``` r
seqtab.nochim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE, verbose=TRUE)
```

    ## Identified 896 bimeras out of 10899 input sequences.

``` r
dim(seqtab.nochim)
```

    ## [1]    20 10003

``` r
sum(seqtab.nochim)/sum(seqtab2)
```

    ## [1] 0.9545064

``` r
taxa <- assignTaxonomy(seqtab.nochim, "~/Desktop/silva_nr_v132_train_set.fa.gz", multithread=TRUE)
```

``` r
saveRDS(taxa, file = "lee_taxa")
```

``` r
saveRDS(seqtab.nochim, file = "lee_trim_seqtab")
```

``` r
tab_l16 <-  readRDS(file="~/Desktop/Ocean-Crust-Synthesis/SRP063681/lee_trim_seqtab")
taxal16 <-readRDS(file="~/Desktop/Ocean-Crust-Synthesis/SRP063681/lee_taxa")
```

``` r
ps_l16 <- phyloseq(otu_table(tab_l16, taxa_are_rows=FALSE), 
               tax_table(taxal16))
ps_l16
```

    ## phyloseq-class experiment-level object
    ## otu_table()   OTU Table:         [ 10003 taxa and 20 samples ]
    ## tax_table()   Taxonomy Table:    [ 10003 taxa by 6 taxonomic ranks ]

``` r
colnames(tax_table(ps_l16)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
```

``` r
saveRDS(ps_l16, file = "~/Desktop/Ocean-Crust-Synthesis/SRP063681/ps_l16")
```
